middleware/authed.js

```diff
@@ -42,34 +73,34 @@ function requireUserAuth(req, res, next) {
  * Middleware to require that the visitor is logged in as a staff user
  */
 function requireStaffAuth(req, res, next) {
-    if (req.isAuthenticated() && isRegularUser(req.user)) {
-        res.redirect('/user');
-    } else if (req.isAuthenticated() && isStaff(req.user)) {
-        return next();
+    if (req.isAuthenticated()) {
+        if (isStaff(req.user)) {
+            next();
+        } else {
+            res.redirect('/user');
+        }
     } else {
         res.redirect(`/user/staff/login?redirectUrl=${req.originalUrl}`);
     }
 }
 
-function redirectUrlWithFallback(fallbackUrl, req, res) {
-    let redirectUrl = fallbackUrl;
-    if (req.query.redirectUrl) {
-        redirectUrl = req.query.redirectUrl;
-    } else if (req.body.redirectUrl) {
-        redirectUrl = req.body.redirectUrl;
-    } else if (req.session.redirectUrl) {
-        redirectUrl = req.session.redirectUrl;
-        delete req.session.redirectUrl;
+/*
+ * Required note staff auth
+ * Middleware to require that the visitor is NOT logged in as a staff user
+ */
+function requireNotStaffAuth(req, res, next) {
+    if (req.isAuthenticated() && isStaff(req.user)) {
+        res.redirect('/tools');
```

Comment:
Wonder if we should show a message to staff here that they tried to access something only meant for regular accounts? eg. append `?s=regularOnly` or something on the URL?